packages/ux-capture/src/Zone.js

```diff
@@ -45,8 +94,37 @@ function Zone(props) {
 	this.marks.forEach(({ mark, measureListener }) => {
 		mark.addOnMarkListener(measureListener);
 	});
+
+	// if expected some marks, but all are already on the page,
+	// create zero-length measure
+	if (configuredMarkNames.length > 0 && markNamesToExpect.length === 0) {
+		this.measure(this.props.startMarkName);
+	}
 }
 
+/**
+ * Returns DOM nodes collection / array or individual DOM Node based on selector configuration
+ *
+ * @param {Object} element - individual element configuration object
+ * @returns {Node|Node[]|null}
+ */
+Zone.prototype.selectDOMNodes = function (element) {
+	// if elements have selectors defined or global element selector is configured,
+	// use them to find marks to record
+	if (element.selector) {
+		if (typeof element.selector === 'function') {
+			return element.selector();
+		} else {
+			// if not a function, treat it as CSS selector argument
+			return document.querySelectorAll(element.selector);
+		}
+	} else if (this.props.selector) {
```

Comment:
is this always a function, or could it be a CSS-style selector string like `element.selector`?

Reply:
This has to be a function as it is set on the zone object and can contain multiple elements - it will be passed an element in question as parameter.